zerocopy-derive/tests/ui/derive_transparent.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+extern crate zerocopy;
+
+fn main() {}
+
+use core::marker::PhantomData;
+use zerocopy::{AsBytes, FromBytes, Unaligned};
+
+// Test generic transparent structs
+
+#[derive(AsBytes, FromBytes, Unaligned)]
+#[repr(transparent)]
+struct TransparentStruct<T> {
+    inner: T,
+    _phantom: PhantomData<()>,
+}
+
+fn assert_is_asbytes<T: AsBytes>() {}
+fn assert_is_frombytes<T: FromBytes>() {}
+fn assert_is_unaligned<T: Unaligned>() {}
+
+struct NotAsBytes;
+
+const _: fn() = assert_is_asbytes::<TransparentStruct<NotAsBytes>>;
+const _: fn() = assert_is_frombytes::<TransparentStruct<char>>;
+const _: fn() = assert_is_unaligned::<TransparentStruct<u32>>;
```

Comment:
Same here - maybe we can merge #118 first and then you can use `AU16`?

Reply:
Yep, sorry missed the update on that PR. I left an approval and will rebase this once it merges.

Reply:
Merged!